lib/daemon.py

```diff
@@ -31,7 +35,17 @@ def main() -> int:
 
     notify("READY=1")
 
-    daemon.poll_for_pro_license(cfg)
+    if os.path.exists("/run/cloud-init/cloud-id-gce") and not os.path.exists(
```

Comment:
Isn't the `should_poll_for_pro_license` already handle that case ? I mean it should only return `True` for `gcp` instances right ?

Reply:
Yes, I just want to be able to look at this file and at the `ubuntu-advantage.service` ConditionPathExists rules, and be able to easily audit that each rule triggers a different background activity.

Without this check here, a reviewer would need deeper knowledge of the `should_poll_for_pro_license` function (like you have :smile: ). I'm also worried about future refactors changing that check and us accidentally ending up in the wrong mode.